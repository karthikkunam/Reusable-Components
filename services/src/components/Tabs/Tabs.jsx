/* eslint-disable react/destructuring-assignment */
/* eslint-disable arrow-body-style */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Tabs, Tab } from 'react-bootstrap';

export class TabsComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeKey: props.activeKey,
      tabs: props.tabs,
    };
  }

  setKey=(k) => {
    this.setState({ activeKey: k });
    this.props.tabKey(k);
  }

  render() {
    const { activeKey, tabs } = this.state;
    return (
      <Tabs id="controlled-tab-example" activeKey={activeKey} onSelect={k => this.setKey(k)} >
        { tabs && tabs.length > 0 && tabs.map((tab) => {
          return (
            <Tab key={tab.eventKey} eventKey={tab.eventKey} title={tab.title} disabled={tab.disabled}/>
          );
        })
        }
      </Tabs>

    );
  }
}
TabsComponent.displayName = 'Tabs';

TabsComponent.propTypes = {
  activeKey: PropTypes.string,
  tabs: PropTypes.arrayOf( 
    PropTypes.shape({ 
      eventKey: PropTypes.string,
      title:  PropTypes.string,
      urlParam:  PropTypes.string,
      disabled: PropTypes.bool
    }) 
  ), 
  tabKey: PropTypes.func,
};

TabsComponent.defaultProps = {
  activeKey: 'DeliveryStatus',
  tabs: [
    {
      eventKey: 'DeliveryStatus',
      title: 'Delivery Status',
      urlParam: 'delivery-status'
    },
    {
      eventKey: 'InvoiceSummary',
      title: 'Invoice Summary',
      urlParam: 'invoice-summary'
    },
    {
      eventKey: 'FinalExceptionReports',
      title: 'Final Exception Reports',
      urlParam: 'final-exception-reports'
    }
  ],
  tabKey: () => {}
};

export default TabsComponent;
