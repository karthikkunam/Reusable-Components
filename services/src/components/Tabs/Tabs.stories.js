import React from 'react';
import { action } from '@storybook/addon-actions';
import TabsComponent from '.';

export default {
    title: 'TabsComponent',
    component: TabsComponent,
};

const disableTab=[
    {
        eventKey: 'DeliveryStatus',
        title: 'Delivery Status',
        urlParam: 'delivery-status'
    },
    {
      eventKey: 'InvoiceSummary',
      title: 'Invoice Summary',
      urlParam: 'invoice-summary'
    },
    {
      eventKey: 'FinalExceptionReports',
      title: 'Final Exception Reports',
      urlParam: 'final-exception-reports',
      disabled: true
    }
];


const enableTab= [
    {
      eventKey: 'DeliveryStatus',
      title: 'Delivery Status',
      urlParam: 'delivery-status'
    },
    {
      eventKey: 'InvoiceSummary',
      title: 'Invoice Summary',
      urlParam: 'invoice-summary'
    },
    {
      eventKey: 'FinalExceptionReports',
      title: 'Final Exception Reports',
      urlParam: 'final-exception-reports'
    }
  ];

  export const ActiveTabs = () => (
    <TabsComponent
    activeKey= 'DeliveryStatus'
    tabs = {enableTab}
    tabKey = {action('Switched tab')}
    />
  );
  export const TabsWithOneDisableTab = () => (
    <>
    <TabsComponent
    activeKey= 'DeliveryStatus'
    tabs = {disableTab}
    tabKey = {action('Switched tab')}
    />
    </>
  );
