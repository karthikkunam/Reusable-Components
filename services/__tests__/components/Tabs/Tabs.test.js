import React from 'react';
import { shallow } from 'enzyme';
import Tabs from '../../../src/components/Tabs';

const props = {
  activeKey: 'DeliveryStatus',
  tabs: [
    {
      eventKey: 'DeliveryStatus',
      title: 'Delivery Status',
      urlParam: 'delivery-status',
    },
    {
      eventKey: 'InvoiceSummary',
      title: 'Invoice Summary',
      urlParam: 'invoice-summary',
    },
    {
      eventKey: 'FinalExceptionReports',
      title: 'Final Exception Reports',
      urlParam: 'final-exception-reports',
    },
  ],
  tabKey: () => {},
};
describe('Tabs Component', () => {
  it('renders correctly', () => {
    const component = shallow(<Tabs {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders setKey when onSelect is triggered', () => {
    const component = shallow(<Tabs {...props} />);
    component
      .find('#controlled-tab-example')
      .at(0)
      .simulate('select', {
        setKey: () => {},
      });
    expect(component).toMatchSnapshot();
  });
});
