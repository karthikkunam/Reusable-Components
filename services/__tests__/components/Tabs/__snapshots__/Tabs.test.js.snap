// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component renders correctly 1`] = `
<Tabs
  activeKey="DeliveryStatus"
  id="controlled-tab-example"
  mountOnEnter={false}
  onSelect={[Function]}
  unmountOnExit={false}
  variant="tabs"
>
  <Tab
    eventKey="DeliveryStatus"
    key="DeliveryStatus"
    title="Delivery Status"
  />
  <Tab
    eventKey="InvoiceSummary"
    key="InvoiceSummary"
    title="Invoice Summary"
  />
  <Tab
    eventKey="FinalExceptionReports"
    key="FinalExceptionReports"
    title="Final Exception Reports"
  />
</Tabs>
`;

exports[`Tabs Component renders setKey when onSelect is triggered 1`] = `
<Tabs
  activeKey={
    Object {
      "setKey": [Function],
    }
  }
  id="controlled-tab-example"
  mountOnEnter={false}
  onSelect={[Function]}
  unmountOnExit={false}
  variant="tabs"
>
  <Tab
    eventKey="DeliveryStatus"
    key="DeliveryStatus"
    title="Delivery Status"
  />
  <Tab
    eventKey="InvoiceSummary"
    key="InvoiceSummary"
    title="Invoice Summary"
  />
  <Tab
    eventKey="FinalExceptionReports"
    key="FinalExceptionReports"
    title="Final Exception Reports"
  />
</Tabs>
`;
